### ISSUE TITLE
UnnecessarySemicolon false positive on comments

### ISSUE THREAD

**Issue by grv87** [CodeNarc/CodeNarc#236]
Apache 2.0 original license notice includes semicolon at the end of the line:
```groovy
/* ...
   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
... */
```

CodeNarc tells that this is a violation of UnnecessarySemicolon.

Documentation says that `excludePattern` should exclude comments. Looks like its default value is not applied or is incomplete.

**Comment by chrismair** [comment]
That rule uses regular expressions to find lines ending with a semicolon, and ignores lines matching several types of comments. So, if those lines had a leading asterisk, it would ignore them, e.g.

```
/*
 * Copyright 2011 the original author or authors.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
```

I acknowledge that is just a workaround, and an annoying one at that.

**Comment by grv87** [comment]
And if it is regular (non-comment) line with leading asterisk, is it ignored?

**Comment by chrismair** [comment]
Yes.

**Comment by grv87** [comment]
@chrismair, maybe CodeNarc should have a rule to check style of multiline comments (i.e. that they have leading asterisk). What do you think?

**Comment by grv87** [comment]
Similar problem with `BlankLineBeforePackage` rule
False positive on blank lines in multi-line comments

**Comment by chrismair** [comment]
> @chrismair, maybe CodeNarc should have a rule to check style of multiline comments (i.e. that they have leading asterisk). 

Yes, that sounds like it could be useful.

**Comment by russellsanborn** [comment]
I submitted a PR that includes a new `MultilineCommentChecker` class that i think could be used not only for solving the issues mentioned here in `BlankLineBeforePackageRule` and `UnnecessarySemicolon`, but could be used for developing a multiline comment style rule.

Hope it helps

**Comment by chrismair** [comment]
This was fixed in CodeNarc 1.1.
